cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfJointComponent.cpp

```diff
@@ -85,19 +86,93 @@ void UUrdfJointComponent::initializeComponent(UrdfJointDesc* joint_desc, UUrdfLi
             ConstraintInstance.SetAngularSwing2Limit(EAngularConstraintMotion::ACM_Free, 0.0f);
             break;
         }
-    case UrdfJointType::Planar:
-        {
+        case UrdfJointType::Planar: {
             float linear_limit_size = (joint_desc->upper_ - joint_desc->lower_) * m_to_cm;
 
             ConstraintInstance.SetLinearXLimit(ELinearConstraintMotion::LCM_Limited, linear_limit_size);
             ConstraintInstance.SetLinearYLimit(ELinearConstraintMotion::LCM_Limited, linear_limit_size);
             break;
         }
-    default:
-        ASSERT(false);
-        break;
+        default: {
+            ASSERT(false);
+            break;
+        }
     }
 
     SetDisableCollision(true);
     SetConstrainedComponents(parent_link_component, NAME_None, child_link_component, NAME_None);
 }
+
+void UUrdfJointComponent::addAction(float action)
```

Comment:
You could maybe implement addAction by calling applyAction. This wouldn't really save any effort now, but it might when you implement torque control. This is just something to think about.

Reply:
Lets leave it as it is and update later if necessary or we find better way to do it. Sadly the current action is different for each case, so merging would not makes the code cleaner.